File: sdks/typescript/src/resources/branches.ts
import { AxiosInstance } from 'axios';
import { PaginatedResponse } from '../types';

export interface Branch {
    id: string;
    name: string;
    address?: string;
    phone?: string;
    email?: string;
    is_active: boolean;
    created_at: string;
    updated_at: string;
}

export class Branches {
    constructor(private client: AxiosInstance) { }

    async list(page: number = 1, pageSize: number = 20): Promise<PaginatedResponse<Branch>> {
        const response = await this.client.get('/branches/', {
            params: { page, page_size: pageSize }
        });
        return response.data;
    }

    async get(branchId: string): Promise<Branch> {
        const response = await this.client.get(`/branches/${branchId}/`);
        return response.data;
    }

    async create(data: Partial<Branch>): Promise<Branch> {
        const response = await this.client.post('/branches/', data);
        return response.data;
    }

    async update(branchId: string, data: Partial<Branch>): Promise<Branch> {
        const response = await this.client.patch(`/branches/${branchId}/`, data);
        return response.data;
    }

    async delete(branchId: string): Promise<void> {
        await this.client.delete(`/branches/${branchId}/`);
    }
}

File: sdks/rust/src/resources/branches.rs
use crate::models::{PaginatedResponse, Branch};
use crate::Puxbay;
use crate::PuxbayError;
use reqwest::Method;

pub struct BranchesResource<'a> {
    pub(crate) client: &'a Puxbay,
}

impl<'a> BranchesResource<'a> {
    pub fn new(client: &'a Puxbay) -> Self {
        Self { client }
    }

    pub async fn list(&self, page: i32) -> Result<PaginatedResponse<Branch>, PuxbayError> {
        self.client.request(
            Method::GET,
            &format!("branches/?page={}", page),
            None::<()>,
        ).await
    }

    pub async fn get(&self, branch_id: &str) -> Result<Branch, PuxbayError> {
        self.client.request(
            Method::GET,
            &format!("branches/{}/", branch_id),
            None::<()>,
        ).await
    }

    pub async fn create(&self, branch: &Branch) -> Result<Branch, PuxbayError> {
        self.client.request(
            Method::POST,
            "branches/",
            Some(branch),
        ).await
    }

    pub async fn update(&self, branch_id: &str, branch: &Branch) -> Result<Branch, PuxbayError> {
        self.client.request(
            Method::PATCH,
            &format!("branches/{}/", branch_id),
            Some(branch),
        ).await
    }

    pub async fn delete(&self, branch_id: &str) -> Result<(), PuxbayError> {
        self.client.request::<serde_json::Value>(
            Method::DELETE,
            &format!("branches/{}/", branch_id),
            None::<()>,
        ).await?;
        Ok(())
    }
}
